grudge/models/__init__.py

```diff
@@ -36,16 +38,24 @@ class Operator:
 class HyperbolicOperator(Operator):
     """A base class for hyperbolic Discontinuous Galerkin operators."""
 
-    def max_eigenvalue(self, t, fields, discr):
-        raise NotImplementedError
+    @abstractmethod
+    def max_eigenvalue(self, t, fields, dcoll):
+        """Return an upperbound on the maximal eigenvalue for the operator."""
```

Comment:
While the characteristic speed are typically found as an eigenvalues, I think a rename to `max_characteristic_vel` would be an improvement.

Reply:
Good idea, I like that suggestion: https://github.com/inducer/grudge/pull/108/commits/6998f984145151df07d85e1e8eaf981cc59092b6